### PR TITLE
IBX-1095: Behat coverage for login scenarios with email provider

### PR DESCRIPTION
**Pull request by micszo**
**JIRA**: https://issues.ibexa.co/browse/IBX-1095

Things done:
- introduced setup scenario modifying security.yaml (adding chain with both username and email providers)
- added support for known (not random) email in createUser
- improved error handling for driver

Related PR: https://github.com/ezsystems/ezplatform-user/pull/103

### CODE REVIEW

File: src/lib/API/Facade/UserFacade.php [hunk 1]
```diff
@@ -63,11 +63,11 @@ public function createUserGroup(string $groupName)
         $this->userService->createUserGroup($userGroupStruct, $parentGroup);
     }
 
-    public function createUser($userName, $userLastName, $userGroupName = null, $languageCode = 'eng-GB')
+    public function createUser($userName, $userLastName, $userGroupName = null, $userEmail = null, $languageCode = 'eng-GB')
     {
         $userCreateStruct = $this->userService->newUserCreateStruct(
             $userName,
-            $this->randomDataGenerator->getFaker()->email,
+            $userEmail == null ? $this->randomDataGenerator->getFaker()->email : $userEmail,
```
Comment (mnocon) [inline src/lib/API/Facade/UserFacade.php:70]: I'd move this outside of the function call

You can most likely write it as:
```
$userEmail = $userEmail ?? $this->randomDataGenerator->getFaker()->email;
$userCreateStruct = $this->userService->newUserCreateStruct(
            $userName,
            $userEmail,
```

BTW for PHP 7.4 we could use:
`$userEmail ??= $this->randomDataGenerator->getFaker()->email;`

**Review by mnocon — APPROVED** [review]
Looks good, there's one thing that's missing for me - please add the setup Scenario to the setup suites of regression profiles (so that these tests are executed during nightly runs).

**Review by piotrbisaga — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]

**Review by bogusez — APPROVED** [review]

**Comment by micszo** [comment]
> please add the setup Scenario to the setup suites of regression profiles (so that these tests are executed during nightly runs).

Verified with regression suite that new tests, including setup scenario, are included, ref. https://app.travis-ci.com/github/ezsystems/ezplatform-page-builder/builds/242566911.

**Comment by mnocon** [comment]
> > please add the setup Scenario to the setup suites of regression profiles (so that these tests are executed during nightly runs).
> 
> Verified with regression suite that new tests, including setup scenario, are included, ref. https://app.travis-ci.com/github/ezsystems/ezplatform-page-builder/builds/242566911.

Great, thanks!

**Review by kacper-wieczorek-ibexa — APPROVED** [review]

**Pull request merged**